shell.nix

```diff
@@ -0,0 +1,78 @@
+with (import <nixpkgs> { });
+let
+  lib = stdenv.lib;
+  buildPythonPackage = python3Packages.buildPythonPackage;
+
+  libsass = buildPythonPackage rec {
+    pname = "libsass";
+    version = "0.20.0";
+
+    src = fetchFromGitHub {
+      owner = "sass";
+      repo = "libsass-python";
+      rev = version;
+      sha256 = "0h9rj4k9izkfdvli8ip72bbvh6a7bvrv5pxz6zay2bq235gpfgfc";
+    };
+
+    buildInputs = [
+      pkgs.libsass
+    ];
+
+    propagatedBuildInputs = [
+      python3Packages.six
+    ];
+
+    preBuild = ''
+      export SYSTEM_SASS=true;
+    '';
+
+    meta = with lib; {
+      description = "A straightforward binding of libsass for Python. Compile Sass/SCSS in Python with no Ruby stack at all!";
+      homepage = "https://sass.github.io/libsass-python/";
+      license = licenses.mit;
+      maintainers = with maintainers; [ SuperSandro2000 ];
+    };
+  };
+
+  lektor-scss = buildPythonPackage rec {
+    pname = "lektor-scss";
+
+    version = "1.3.8";
+
+    src = fetchFromGitHub {
+      owner = "chaos-bodensee";
+      repo = "lektor-scss";
+      rev = version;
+      sha256 = "1j5ikjxhg9d9cb4qdvqjdjr861i4ha7h3q52fk1q8wfpzkb451ds";
+    };
+
+    propagatedBuildInputs = [
+      libsass
+      python3Packages.termcolor
+    ];
+
+    postPatch = ''
+      sed -i "s/libsass==0.20.0/libsass~=0.20.0/" setup.py
+    '';
+
+    meta = with lib; {
+      description = "Lektor plugin to compile css out of sass - based on libsass";
+      homepage = "https://pypi.org/project/lektor-scss/";
+      license = licenses.mit;
+      maintainers = with maintainers; [ SuperSandro2000 ];
+    };
+  };
+
+  customPython = pkgs.python3.buildEnv.override {
+    extraLibs = with pkgs.python3Packages; [
+      lektor
+      lektor-scss
+      libsass
+      pip
+    ];
+  };
+in
+mkShell
+{
+  buildInputs = [ customPython ];
+}
```

Comment:
Unfortunately I don't use Nix yet, so excuse the question.

How does it work with this `shell.nix` file? It seems to install the lektor-scss plugin as well.

How would the way be to install lektor together with lektor-scss under Nix and then start the local webserver ``make server`` or ``lektor server -h 127.0.0.1 -f scss``?

Could it be that we want to put the Nix part into the make install part of the Makefile?

Reply:
You can just run ``nix-shell``. It downloads and builds the right python packages, stores them in the nix store and enables them for this shell. If you close it again nothing is enabled and everything is in the nix store. When you garbage collect everything unreferenced is deleted.

I tested it locally by running ``nix-shell`` and then ``make server``.

We can easily execute a command in the nix-shell like ``nix-shell --command 'pip list'`` or ``nix-shell --command 'make build'``. We probably want to test if ``nix-shell`` is available because you can install it on other OS's than nixOS like Debian.

Reply:
thanks for explaining :heart: